### PR TITLE
Reset bool config by checking type first

### PR DESCRIPTION
**Pull request by ashione**
## Related issue number
https://github.com/ray-project/ray/issues/9246

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by ashione** [comment]
I have create the following example code :
```cpp
#include <sstream>
#include <iostream>
#include <string>
using namespace std;

template<typename T>
T ResetValue(std::string t, T v) {
    T flag = v;
    if (typeid(T) == typeid(bool)) {
      flag = t == "true";
    } else {
      std::istringstream stream(t);
      stream >> flag;
    }
    return flag;
}

int main () {
  cout << ResetValue("true", false) << endl;
  cout << ResetValue("2", 1) << endl;
  return 0;
}
```
Outputs are 1 and 2.

**Comment by ashione** [comment]
@raulchen @zhijunfu Could you verify this PR or any better suggestions?

File: src/ray/common/ray_config.h [hunk 1]
```diff
@@ -50,8 +51,12 @@ class RayConfig {
 /// A helper macro that helps to set a value to a config item.
 #define RAY_CONFIG(type, name, default_value) \
   if (pair.first == #name) {                  \
-    std::istringstream stream(pair.second);   \
-    stream >> name##_;                        \
+    if (typeid(type) == typeid(bool)) {       \
+       name##_ = pair.second == "true";       \
```
Comment (raulchen) [inline src/ray/common/ray_config.h:55]: covert `pair.second` to lowercase

**Review by raulchen — APPROVED** [review]

**Review by zhijunfu — APPROVED** [review]

File: src/ray/common/ray_config.h (hunk 1 above)
Comment (ashione) [reply src/ray/common/ray_config.h]: > covert `pair.second` to lowercase

Done.

**Pull request merged**